kordex/src/main/kotlin/dev/kordex/gradle/plugins/kordex/extensions/KordExExtension.kt

```diff
@@ -7,20 +7,23 @@
 package dev.kordex.gradle.plugins.kordex.extensions
 
 import dev.kordex.gradle.plugins.kordex.DataCollection
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+import org.gradle.kotlin.dsl.assign
 
-open class KordExExtension {
-	var mainClass: String? = null
+interface KordExExtension {
```

Comment:
Switching to `Property` significantly changes the way I have to document the API for this plugin, and it makes the plugin code significantly less readable - why is it preferred over simple properties?

Reply:
It's part of an ongoing effort to eliminate the after evaluate function, due to the evaluation of it being lazy, it's also useful for configuration caching

Also the API should be almost identical I don't see how the documentation would change in a significant way

Reply:
Can you store `null` in a non-nullable Property?

Reply:
I mean, how do you reset a Property value, so there's no longer anything stored in it? Nullable fields were nullable for a reason before, after all.

Reply:
[Property<T>#unset()](https://docs.gradle.org/current/javadoc/org/gradle/api/provider/Property.html#unset())